internal/services/servicebus/internal.go

```diff
@@ -92,6 +92,18 @@ func authorizationRuleSchemaFrom(s map[string]*pluginsdk.Schema) map[string]*plu
 			Computed:  true,
 			Sensitive: true,
 		},
+
+		"alias_primary_connection_string": {
+			Type:      pluginsdk.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+
+		"alias_secondary_connection_string": {
```

Comment:
could we change these to
```suggestion
		"primary_connection_string_alias": {
			Type:      pluginsdk.TypeString,
			Computed:  true,
			Sensitive: true,
		},

		"secondary_connection_string_alias": {
```

Reply:
hey @katbyte sure will do those changes on naming. 

Regarding the tests, these resource types requires a primary and secondary namespace be in a paired state & they are in a 'transitioning' state for a while as the geo-dr operation is initiated - I found they're a bit temperamental when run in parallel.  I had to run one at a time to get a consistent result, i.e.

![image](https://user-images.githubusercontent.com/5038454/129487005-8f4d84a4-fec0-44e0-a116-c9efb1c7d50f.png)

...but, this _is_ consistent on multiple runs. We wait for [state change to be success](https://github.com/hashicorp/terraform-provider-azurerm/blob/cc272dc609ae10dcc13aa514c81427f454c07d00/internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go#L304) before returning from the DR operation & I tried to control ordering using a depends_on approach for the auth rule, but ran into 'BreakPairing operation....' issue during destroy operations [as commented in the Event Hubs testing implementation](https://github.com/hashicorp/terraform-provider-azurerm/blob/b0c897055329438be6a3a159f6ffac4e1ce958f2/internal/services/eventhub/eventhub_namespace_authorization_rule_data_source_test.go#L39). 

When you make changes to the primary of a paired resource, it has to replicate those changes to the secondary - it's in a 'replicating' state for a while and doesn't accept new operations until complete. I think this is the source of the 'BreakPairing' issue - we're removing the auth rule, ASB is replicating that change but we're not waiting for that to complete so it's immediately trying to break the pair and request is rejected.

Perhaps what we need here is actually a few changes to the authorization rule resources to also wait for replication to complete before returning, which would let us use the depends_on approach from tests.....let me have a look at that.

Reply:
OK  I've added a method ```waitForPairedNamespaceReplication``` which is executed as part of authorization rule creates or deletes. This now works with depends_on, which I've added to the tests - looks good. 
```
$ ./tests.sh
==> Running all tests...
=== RUN   TestAccDataSourceServiceBusNamespaceAuthorizationRule_basic
=== PAUSE TestAccDataSourceServiceBusNamespaceAuthorizationRule_basic
=== RUN   TestAccDataSourceServiceBusNamespaceAuthorizationRule_withAliasConnectionString
=== PAUSE TestAccDataSourceServiceBusNamespaceAuthorizationRule_withAliasConnectionString
=== RUN   TestAccServiceBusNamespaceAuthorizationRule_listen
=== PAUSE TestAccServiceBusNamespaceAuthorizationRule_listen
=== RUN   TestAccServiceBusNamespaceAuthorizationRule_send
=== PAUSE TestAccServiceBusNamespaceAuthorizationRule_send
=== RUN   TestAccServiceBusNamespaceAuthorizationRule_listensend
=== PAUSE TestAccServiceBusNamespaceAuthorizationRule_listensend
=== RUN   TestAccServiceBusNamespaceAuthorizationRule_manage
=== PAUSE TestAccServiceBusNamespaceAuthorizationRule_manage
=== RUN   TestAccServiceBusNamespaceAuthorizationRule_rightsUpdate
=== PAUSE TestAccServiceBusNamespaceAuthorizationRule_rightsUpdate
=== RUN   TestAccServiceBusNamespaceAuthorizationRule_requiresImport
=== PAUSE TestAccServiceBusNamespaceAuthorizationRule_requiresImport
=== RUN   TestAccDataSourceServiceBusQueueAuthorizationRule_basic
=== PAUSE TestAccDataSourceServiceBusQueueAuthorizationRule_basic
=== RUN   TestAccDataSourceServiceBusQueueAuthorizationRule_withAliasConnectionString
=== PAUSE TestAccDataSourceServiceBusQueueAuthorizationRule_withAliasConnectionString
=== RUN   TestAccServiceBusQueueAuthorizationRule_listen
=== PAUSE TestAccServiceBusQueueAuthorizationRule_listen
=== RUN   TestAccServiceBusQueueAuthorizationRule_send
=== PAUSE TestAccServiceBusQueueAuthorizationRule_send
=== RUN   TestAccServiceBusQueueAuthorizationRule_listensend
=== PAUSE TestAccServiceBusQueueAuthorizationRule_listensend
=== RUN   TestAccServiceBusQueueAuthorizationRule_manage
=== PAUSE TestAccServiceBusQueueAuthorizationRule_manage
=== RUN   TestAccServiceBusQueueAuthorizationRule_rightsUpdate
=== PAUSE TestAccServiceBusQueueAuthorizationRule_rightsUpdate
=== RUN   TestAccServiceBusQueueAuthorizationRule_requiresImport
=== PAUSE TestAccServiceBusQueueAuthorizationRule_requiresImport
=== RUN   TestAccServiceBusQueueAuthorizationRule_withAliasConnectionString
=== PAUSE TestAccServiceBusQueueAuthorizationRule_withAliasConnectionString
=== RUN   TestAccDataSourceServiceBusTopicAuthorizationRule_basic
=== PAUSE TestAccDataSourceServiceBusTopicAuthorizationRule_basic
=== RUN   TestAccDataSourceServiceBusTopicAuthorizationRule_withAliasConnectionString
=== PAUSE TestAccDataSourceServiceBusTopicAuthorizationRule_withAliasConnectionString
=== RUN   TestAccServiceBusTopicAuthorizationRule_listen
=== PAUSE TestAccServiceBusTopicAuthorizationRule_listen
=== RUN   TestAccServiceBusTopicAuthorizationRule_send
=== PAUSE TestAccServiceBusTopicAuthorizationRule_send
=== RUN   TestAccServiceBusTopicAuthorizationRule_listensend
=== PAUSE TestAccServiceBusTopicAuthorizationRule_listensend
=== RUN   TestAccServiceBusTopicAuthorizationRule_manage
=== PAUSE TestAccServiceBusTopicAuthorizationRule_manage
=== RUN   TestAccServiceBusTopicAuthorizationRule_requiresImport
=== PAUSE TestAccServiceBusTopicAuthorizationRule_requiresImport
=== RUN   TestAccServiceBusTopicAuthorizationRule_rightsUpdate
=== PAUSE TestAccServiceBusTopicAuthorizationRule_rightsUpdate
=== RUN   TestAccServiceBusTopicAuthorizationRule_withAliasConnectionString
=== PAUSE TestAccServiceBusTopicAuthorizationRule_withAliasConnectionString
=== CONT  TestAccDataSourceServiceBusNamespaceAuthorizationRule_basic
=== CONT  TestAccServiceBusQueueAuthorizationRule_rightsUpdate
=== CONT  TestAccServiceBusNamespaceAuthorizationRule_requiresImport
=== CONT  TestAccServiceBusNamespaceAuthorizationRule_listensend
--- PASS: TestAccDataSourceServiceBusNamespaceAuthorizationRule_basic (295.82s)
=== CONT  TestAccServiceBusQueueAuthorizationRule_manage
--- PASS: TestAccServiceBusNamespaceAuthorizationRule_listensend (296.02s)
=== CONT  TestAccServiceBusQueueAuthorizationRule_listensend
--- PASS: TestAccServiceBusQueueAuthorizationRule_rightsUpdate (366.19s)
=== CONT  TestAccServiceBusQueueAuthorizationRule_send
--- PASS: TestAccServiceBusNamespaceAuthorizationRule_requiresImport (474.45s)
=== CONT  TestAccServiceBusQueueAuthorizationRule_listen
--- PASS: TestAccServiceBusQueueAuthorizationRule_manage (276.76s)
=== CONT  TestAccDataSourceServiceBusQueueAuthorizationRule_withAliasConnectionString
--- PASS: TestAccServiceBusQueueAuthorizationRule_listensend (292.75s)
=== CONT  TestAccDataSourceServiceBusQueueAuthorizationRule_basic
--- PASS: TestAccServiceBusQueueAuthorizationRule_send (285.56s)
=== CONT  TestAccServiceBusNamespaceAuthorizationRule_send
--- PASS: TestAccServiceBusQueueAuthorizationRule_listen (297.42s)
=== CONT  TestAccServiceBusNamespaceAuthorizationRule_listen
--- PASS: TestAccDataSourceServiceBusQueueAuthorizationRule_basic (301.41s)
=== CONT  TestAccDataSourceServiceBusNamespaceAuthorizationRule_withAliasConnectionString
--- PASS: TestAccServiceBusNamespaceAuthorizationRule_send (268.03s)
=== CONT  TestAccServiceBusNamespaceAuthorizationRule_rightsUpdate
--- PASS: TestAccServiceBusNamespaceAuthorizationRule_listen (272.19s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_listensend
--- PASS: TestAccServiceBusNamespaceAuthorizationRule_rightsUpdate (357.54s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_withAliasConnectionString
--- PASS: TestAccServiceBusTopicAuthorizationRule_listensend (283.81s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_rightsUpdate
--- PASS: TestAccDataSourceServiceBusQueueAuthorizationRule_withAliasConnectionString (911.38s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_requiresImport
--- PASS: TestAccServiceBusTopicAuthorizationRule_requiresImport (294.24s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_manage
--- PASS: TestAccServiceBusTopicAuthorizationRule_rightsUpdate (576.45s)
=== CONT  TestAccDataSourceServiceBusTopicAuthorizationRule_withAliasConnectionString
--- PASS: TestAccDataSourceServiceBusNamespaceAuthorizationRule_withAliasConnectionString (1014.82s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_send
--- PASS: TestAccServiceBusTopicAuthorizationRule_manage (272.51s)
=== CONT  TestAccServiceBusTopicAuthorizationRule_listen
--- PASS: TestAccServiceBusTopicAuthorizationRule_send (272.85s)
=== CONT  TestAccServiceBusNamespaceAuthorizationRule_manage
--- PASS: TestAccServiceBusTopicAuthorizationRule_withAliasConnectionString (928.78s)
=== CONT  TestAccServiceBusQueueAuthorizationRule_withAliasConnectionString
--- PASS: TestAccServiceBusTopicAuthorizationRule_listen (246.48s)
=== CONT  TestAccDataSourceServiceBusTopicAuthorizationRule_basic
--- PASS: TestAccServiceBusNamespaceAuthorizationRule_manage (295.86s)
=== CONT  TestAccServiceBusQueueAuthorizationRule_requiresImport
--- PASS: TestAccDataSourceServiceBusTopicAuthorizationRule_basic (276.74s)
--- PASS: TestAccServiceBusQueueAuthorizationRule_requiresImport (295.88s)
--- PASS: TestAccDataSourceServiceBusTopicAuthorizationRule_withAliasConnectionString (956.48s)
--- PASS: TestAccServiceBusQueueAuthorizationRule_withAliasConnectionString (874.51s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus    3099.045s
==> Done.

```